### DISCUSSION TITLE
Possible Bug: Wrong Date displayed in frontend

### DISCUSSION THREAD

**Discussion by Ph-St** [photoprism/photoprism#4044] [Q&A] [answered]
1. What is not working as documented?

Importing files I previously downloaded from iCloud, photoprism displays the upload date rather than the date of recording.

2. How can we reproduce it?

It doesn't seem to happen for all files, but I have a large number of files video files for which it does happen. I'm happy to share if that helps. 

Here is screenshot of photoprism frontend:

![bugreport1](https://github.com/photoprism/photoprism/assets/19529376/9229b2f8-1f49-4172-a588-6a68fef730d0)

Here's a snippet from exiftool's report:

![bugreport2](https://github.com/photoprism/photoprism/assets/19529376/638f3c81-d7a1-4ca6-a28d-fec05a0f73a0)

As you can see, "Media Create Date" shows the date the frontend presents, but this is the date the file was created. But Exiftool also records the original creation date in the attribute "Date/Time Original".

3. What behavior do you expect?

Photoprism should display not the file creation date, but the recording date, which is available in the metadata.

Should I file this as a bug report?

**Comment by lastzero** [comment]
Before you proceed, could you please post the full data as shown by the `exiftool -j -n [filename]` command?

Edit: Also check if there are any related sidecar files, e.g. `*.xmp` files or other media files with the same name (except the extension).

**Reply by Ph-St** [reply]
Thanks for your quick reply! Here's the output of exiftool -j -n 20240131_155041_4F501C44.mp4:

[{
  "SourceFile": "20240131_155041_4F501C44.mp4",
  "ExifToolVersion": 12.70,
  "FileName": "20240131_155041_4F501C44.mp4",
  "Directory": ".",
  "FileSize": 6158257,
  "FileModifyDate": "2024:01:31 18:40:10+01:00",
  "FileAccessDate": "2024:01:31 18:40:10+01:00",
  "FileInodeChangeDate": "2024:01:31 21:04:38+01:00",
  "FilePermissions": 100644,
  "FileType": "MP4",
  "FileTypeExtension": "MP4",
  "MIMEType": "video/mp4",
  "MajorBrand": "mp42",
  "MinorVersion": "0.0.1",
  "CompatibleBrands": ["isom","mp41","mp42"],
  "MovieHeaderVersion": 0,
  "CreateDate": "2024:01:31 15:50:41",
  "ModifyDate": "2024:01:31 15:50:43",
  "TimeScale": 600,
  "Duration": 14.0666666666667,
  "PreferredRate": 1,
  "PreferredVolume": 1,
  "PreviewTime": 0,
  "PreviewDuration": 0,
  "PosterTime": 0,
  "SelectionTime": 0,
  "SelectionDuration": 0,
  "CurrentTime": 0,
  "NextTrackID": 6,
  "TrackHeaderVersion": 0,
  "TrackCreateDate": "2024:01:31 15:50:41",
  "TrackModifyDate": "2024:01:31 15:50:43",
  "TrackID": 1,
  "TrackDuration": 14.0633333333333,
  "TrackLayer": 0,
  "TrackVolume": 1,
  "Balance": 0,
  "AudioFormat": "mp4a",
  "AudioChannels": 2,
  "AudioBitsPerSample": 16,
  "AudioSampleRate": 44100,
  "ImageWidth": 1280,
  "ImageHeight": 720,
  "GraphicsMode": 0,
  "OpColor": "0 0 0",
  "CompressorID": "avc1",
  "SourceImageWidth": 1280,
  "SourceImageHeight": 720,
  "XResolution": 72,
  "YResolution": 72,
  "BitDepth": 24,
  "ColorProfiles": "nclx",
  "ColorPrimaries": 1,
  "TransferCharacteristics": 1,
  "MatrixCoefficients": 1,
  "VideoFullRangeFlag": 0,
  "VideoFrameRate": 30,
  "Warning": "[minor] The ExtractEmbedded option may find more tags in the media data",
  "MatrixStructure": "1 0 0 0 1 0 0 0 1",
  "ContentDescribes": 2,
  "MediaHeaderVersion": 0,
  "MediaCreateDate": "2024:01:31 15:50:41",
  "MediaModifyDate": "2024:01:31 15:50:43",
  "MediaTimeScale": 600,
  "MediaDuration": 14.065,
  "MediaLanguageCode": "und",
  "HandlerType": "meta",
  "HandlerDescription": "Core Media Metadata",
  "MetaFormat": "mebx",
  "DateTimeOriginal": "2024:01:21 14:13:17+01:00",
  "MediaDataSize": 6145702,
  "MediaDataOffset": 12555,
  "ImageSize": "1280 720",
  "Megapixels": 0.9216,
  "AvgBitrate": 3495186,
  "Rotation": 90
}]

There are two files in the sidecar folder with the same name (except extension):

20240131_155041_4F501C44.yml
20240131_155041_4F501C44.mp4.jpg

Do you need exiftool outputs for these files, too?

**Reply by graciousgrey** [reply]
Could you run `exiftool -j -g -n filname` for the mp4 file?

Please note that for mp4 files we prefer one of the CreateDates (I would have to look up exactly which one in the code) over the DateTimeOriginal, as this is by definition the field in which the recording date (in UTC) is saved in mp4 files. Most mp4 files do not have a DateTimeOriginal.

**Reply by Ph-St** [reply]
Thanks for your answer, below's the output you asked for. I don't know how this could be solved, but as I said, I have quite a bunch of files for which photoprism shows a date that is in fact a file creation and as such does not reflect at all what the date presentation in photoprism is supposed the show. Would it perhaps be an option to take the DateTimeOriginal in case it is present and only revert to a Creation Date if there's no such date?

exiftool -j -g -n 20240131_155041_4F501C44.mp4
[{
  "SourceFile": "20240131_155041_4F501C44.mp4",
  "ExifTool": {
    "ExifToolVersion": 12.70,
    "Warning": "[minor] The ExtractEmbedded option may find more tags in the media data"
  },
  "File": {
    "FileName": "20240131_155041_4F501C44.mp4",
    "Directory": ".",
    "FileSize": 6158257,
    "FileModifyDate": "2024:01:31 18:40:10+01:00",
    "FileAccessDate": "2024:01:31 18:40:10+01:00",
    "FileInodeChangeDate": "2024:01:31 21:04:38+01:00",
    "FilePermissions": 100644,
    "FileType": "MP4",
    "FileTypeExtension": "MP4",
    "MIMEType": "video/mp4"
  },
  "QuickTime": {
    "MajorBrand": "mp42",
    "MinorVersion": "0.0.1",
    "CompatibleBrands": ["isom","mp41","mp42"],
    "MovieHeaderVersion": 0,
    "CreateDate": "2024:01:31 15:50:41",
    "ModifyDate": "2024:01:31 15:50:43",
    "TimeScale": 600,
    "Duration": 14.0666666666667,
    "PreferredRate": 1,
    "PreferredVolume": 1,
    "PreviewTime": 0,
    "PreviewDuration": 0,
    "PosterTime": 0,
    "SelectionTime": 0,
    "SelectionDuration": 0,
    "CurrentTime": 0,
    "NextTrackID": 6,
    "TrackHeaderVersion": 0,
    "TrackCreateDate": "2024:01:31 15:50:41",
    "TrackModifyDate": "2024:01:31 15:50:43",
    "TrackID": 1,
    "TrackDuration": 14.0633333333333,
    "TrackLayer": 0,
    "TrackVolume": 1,
    "Balance": 0,
    "AudioFormat": "mp4a",
    "AudioChannels": 2,
    "AudioBitsPerSample": 16,
    "AudioSampleRate": 44100,
    "ImageWidth": 1280,
    "ImageHeight": 720,
    "GraphicsMode": 0,
    "OpColor": "0 0 0",
    "CompressorID": "avc1",
    "SourceImageWidth": 1280,
    "SourceImageHeight": 720,
    "XResolution": 72,
    "YResolution": 72,
    "BitDepth": 24,
    "ColorProfiles": "nclx",
    "ColorPrimaries": 1,
    "TransferCharacteristics": 1,
    "MatrixCoefficients": 1,
    "VideoFullRangeFlag": 0,
    "VideoFrameRate": 30,
    "MatrixStructure": "1 0 0 0 1 0 0 0 1",
    "ContentDescribes": 2,
    "MediaHeaderVersion": 0,
    "MediaCreateDate": "2024:01:31 15:50:41",
    "MediaModifyDate": "2024:01:31 15:50:43",
    "MediaTimeScale": 600,
    "MediaDuration": 14.065,
    "MediaLanguageCode": "und",
    "HandlerType": "meta",
    "HandlerDescription": "Core Media Metadata",
    "MetaFormat": "mebx",
    "DateTimeOriginal": "2024:01:21 14:13:17+01:00",
    "MediaDataSize": 6145702,
    "MediaDataOffset": 12555
  },
  "Composite": {
    "ImageSize": "1280 720",
    "Megapixels": 0.9216,
    "AvgBitrate": 3495186,
    "Rotation": 90
  }
}]

**Reply by lastzero** [reply, marked as answer]
I have found DateTimeOriginal listed as a QuickTime UserData field in the Exiftool documentation:

- https://exiftool.org/TagNames/QuickTime.html#Keys

However, the compatible brands are "isom", "mp41" and "mp42" (ISO/MP4), not "qt" (QuickTime), and I would generally assume that it is best to use the fields specified by the MP4 standard rather than a UserData field, which is not standardized and for which - from what I have read - there is also no clear specification when it comes to the timezone and offset used. If you have any further information and know where this is commonly used (e.g. which applications read and write it), please let us know!

**Reply by Ph-St** [reply]
For the MP4, I can give you a pretty complete account of which applications touched it:

1. I've recorded it with an iphone SE2020, ios 16.6.1, with the standard camera app.
2. I've sent the video to a contact via whatsapp.
3. The video was synced to my icloud.
4. I've downloaded the video from icloud with the bulk download option (the files get zipped) with Firefox.
5. I've unzipped the file on my Arch Linux.
6. I've imported the files into photoprism using the photoprism interface with Firefox.
7. Photoprism's storage folder is also synced to a nextcloud instance.

**Reply by Ph-St** [reply]
Contrary to what I originally thought, almost all of the affected files are actually video files, not  jpeg, as I previously thought. I edited my original post to reflect this.
